### ISSUE TITLE
No utilization metrics reported in metrics API

### ISSUE THREAD

**Issue by askfongjojo** [oxidecomputer/omicron#4337]
On rack3, after updating from v1.0.2 to v3, the utilization metrics for cpu, memory and disk are all zero on the web console. The result set is empty when querying directly using the `/v1/metrics` API endpoint, e.g.
```
oxide api '/v1/system/metrics/cpus_provisioned?start_time=2023-10-21T01%3A23%3A14.591Z&end_time=2023-10-25T23%3A23%3A14.591Z&limit=3000'
{
  "items": [],
  "next_page": null
}
```

These are the 3 nexus metrics producers:
```
root@[fd00:1122:3344:116::3]:32221/omicron> select * from metric_producer where port = '12221';
                   id                  |         time_created          |         time_modified         |          ip           | port  | interval |    base_route    |             oximeter_id
---------------------------------------+-------------------------------+-------------------------------+-----------------------+-------+----------+------------------+---------------------------------------
  1cfdb5b6-e568-436a-a85f-7fecf1b8eef2 | 2023-09-02 13:09:15.738668+00 | 2023-10-24 21:56:48.399475+00 | fd00:1122:3344:10e::3 | 12221 |       10 | /metrics/collect | e9826cdc-6d3a-4eff-b1b5-ec4364ebe6b9
  44c35566-dd64-4e4a-896e-c50aaa3df14f | 2023-09-02 13:09:21.049013+00 | 2023-10-24 21:58:11.164166+00 | fd00:1122:3344:111::3 | 12221 |       10 | /metrics/collect | e9826cdc-6d3a-4eff-b1b5-ec4364ebe6b9
  95ebe94d-0e68-421d-9260-c30bd7fe4bd6 | 2023-09-02 13:09:20.946861+00 | 2023-10-24 21:55:46.088914+00 | fd00:1122:3344:108::3 | 12221 |       10 | /metrics/collect | e9826cdc-6d3a-4eff-b1b5-ec4364ebe6b9
(3 rows)
```

The oximeter and nexus log files can be found in catacomb, under `/data/staff/rack3` prefixed with this ticket number.

**Comment by david-crespo** [comment]
It is deliberate that this update clears out Clickhouse.

**Comment by bnaecker** [comment]
Yes, but there are now instances running, so the data in ClickHouse should be non empty. Is it rounded to zero somewhere before display?

**Comment by david-crespo** [comment]
Looks like she hit the API and the list came back empty, so there are no data points. I think clickhouse would only get an entry when the stats _change_, i.e., when an instance changes state in a way that affects that stats.

**Comment by askfongjojo** [comment]
So, it would trigger the repopulation of the metrics _after_ provisioning something new.
I've been holding off from making new instances to see what else I can find. This just happened after I provisioned a new one:
<img width="1266" alt="image" src="https://github.com/oxidecomputer/omicron/assets/10874909/8b821e05-b376-4ca3-bede-abce8e9b30c8">

The thing that is still unexplained is why starting an existing instance didn't do that.

**Comment by bnaecker** [comment]
Oh, so these are gauges and only produced when values change? It seems like we want to report the current value in all collection intervals.

**Comment by david-crespo** [comment]
Yes, I believe we only write to clickhouse when we update the state table in CRDB.

**Comment by david-crespo** [comment]
> So, it would trigger the repopulation of the metrics after provisioning something new.

I think I saw a version of this. I had an instance from before the update and it was stopped. When I started it, there was no update to the stats. When I then stopped it, however, the stats did go down by the expected 64 GiB memory.

<img width="417" alt="image" src="https://github.com/oxidecomputer/omicron/assets/3612203/54df642e-8a26-449d-9421-8632dde1f0a2">
